packages/nimble-components/src/rich-text/specs/README.md

```diff
@@ -126,13 +132,19 @@ _Props/Attrs_
 -   `error-visible` - is a boolean attribute used to visually change the component's border color with the error exclamation at the top right, indicating that an error has occurred, as per the current
     [visual design](https://www.figma.com/file/PO9mFOu5BCl8aJvFchEeuN/Nimble_Components?type=design&node-id=2482-82389&mode=design&t=KwADu9QRoL7QAuIW-0)
 -   `error-text` - is a string attribute that displays the error text at the bottom of the component when the `error-visible` is enabled.
+-   `mentioned-users` - is a read-only property that returns an array of strings representing the mentioned user IDs in the current state of the editor.
+-   `validity` - is a readonly object of boolean values that represents the validity state that the `@mention` configuration can be. The object type
+    is `RichTextMentionValidity`. The validation is especially for mapping the user details that is provided via the
+    `nimble-mapping-mention-user`. For example, if the client application provide the duplicate `key` values that stores the user ID, it will be an
+    issue in scenarios when parsing the mention user from just an user ID to username.
```

Comment:
Part of the goal of the separate configuration elements is so that there is a decoupling from the specific type of mentions and the rich-text-editor / rich-text-viewer (the rich text components can have the concept of mentions but they shouldn't know about all possible mentions, why would a generic rich text control know about users, systems, etc).

So I think the configuration element themselves, i.e. `nimble-rich-text-mention-users` , should have:
- the exposed properties of `mentionedUsers` (similar to how table columns have their currentPixelWidth, etc.)
- the different types of mention validity (invalidURL, duplicateURL, etc.) (similar to how table columns have their own ColumnValidity)
- and fire their own `mention-update` event (similar to how table columns fire `delegated-event`)

Reply:
Agreed. I moved all these APIs as part of `nimble-rich-text-mention-users` instead of a `nimble-rich-text-editor`.